tls/s2n_signature_scheme.h

```diff
@@ -28,6 +28,7 @@ struct s2n_signature_scheme {
     s2n_signature_algorithm sig_alg;
     uint8_t minimum_protocol_version;
     uint8_t maximum_protocol_version;
+    uint16_t libcrypto_nid;
```

Comment:
"libcrypto_nid" isn't really what this is. I'd call it "asn1_nid".

Reply:
Oh and you might want a comment describing how to determine the correct nid, for when future us need to add new signature schemes.

Reply:
I called it that because I'm following a precendant in this file: https://github.com/awslabs/s2n/blob/main/crypto/s2n_ecc_evp.c#L45. I would prefer to keep this consistent if possible.